tsdb/index/tsi1/partition.go

```diff
@@ -348,22 +349,39 @@ func (p *Partition) buildSeriesSet() error {
 }
 
 // CurrentCompactionN returns the number of compactions currently running.
-func (p *Partition) CurrentCompactionN() int {
-	p.mu.RLock()
-	defer p.mu.RUnlock()
-	return p.currentCompactionN
+func (p *Partition) CurrentCompactionN() int32 {
+	return p.currentCompactionN.Load()
 }
 
 // Wait will block until all compactions are finished.
 // Must only be called while they are disabled.
 func (p *Partition) Wait() {
 	ticker := time.NewTicker(10 * time.Millisecond)
 	defer ticker.Stop()
+
+	// Debug level timeout
+	timeoutDuration := 24 * time.Hour
+	startTime := time.Now()
+	var lastWarningTime time.Time
+
 	for {
 		if p.CurrentCompactionN() == 0 {
 			return
 		}
-		<-ticker.C
+		select {
+		case <-ticker.C:
+			elapsed := time.Since(startTime)
```

Comment:
It seems like `lastWarningTime` and `startTime` could be collapsed to a singe variable for simplicity.
```
lastTime := time.Now()
for {
		if p.CurrentCompactionN() == 0 {
			return
		}
		select {
		case <-ticker.C:
			elapsed := time.Since(lastTime)
                        if elapsed >= timeoutDuration {
                                lastTime = time.Now()
				files := make([]string, 0)
				for _, v := range p.fileSet.Files() {
					files = append(files, v.Path())
```